udata/api/oauth2.py

```diff
@@ -255,17 +282,17 @@ def request_invalid(self, request):
         return False
 
     def token_revoked(self, token):
-        # TODO: return token.revoked
-        return False
+        return token.revoked
 
 
 @blueprint.route('/token', methods=['POST'], localize=False, endpoint='token')
 @csrf.exempt
 def access_token():
-    return oauth.create_token_response()
+    return oauth.create_token_response(request)
```

Comment:
I think we should pre-chew the client's request before `create_token_response` so we could allow json data AND form data. Otherwise the flask implementation of Authlib will only accept form data and will ignore json data. That could be problematic to understand from a front point of view and counter-intuitive for third-party uses.

### proposal : 

```
from authlib.oauth2.rfc6749 import OAuth2Request
from authlib.integrations.flask_helpers import create_oauth_request

...

@blueprint.route('/token', methods=['POST'], localize=False, endpoint='token')
@csrf.exempt
def access_token():
    has_form = bool(request.form)
    req = create_oauth_request(request, OAuth2Request, use_json=not has_form)
    return oauth.create_token_response(request=req)
```

... that said if we want 3rd parties to send auth data only as form data (and not json) we could use `return oauth.create_token_response(request)` alone, but I think we should at least update the udata's docs and precise it in an "oauth2 / pkce" section...

### references / explanations from authlib : 

I tested various way before proposing this, because `create_oauth_request` was already integrated in the Flask's `AuthorizationServer` class, and it seemed weird indeed... But after exploring the authlib code I couldn't see where json data could be proceeded : 

- `AuthorizationServer` from Flask implementation does not contain/overwrite `create_token_response` : https://github.com/lepture/authlib/blob/master/authlib/integrations/flask_oauth2/authorization_server.py
- So the `create_token_response` function in Flask implementation's `AuthorizationServer` inherits from `rfc6749`'s `AuthorizationServer` and should pass on a request arg : https://github.com/lepture/authlib/blob/782a0fced780849418dc2a869528d10387e24b65/authlib/oauth2/rfc6749/authorization_server.py#L203
- the request is created from the class's own `create_oauth_request`, but never checks if json data is contained or not : https://github.com/lepture/authlib/blob/782a0fced780849418dc2a869528d10387e24b65/authlib/integrations/flask_oauth2/authorization_server.py#L110
- note that the json data from the client's request is only injected into an `OAuth2Request` instance if `use_json == true`, and none of the calls I've seen so far in the Flask implementation trigger this kwarg (but perhaps I've missed something ?) : https://github.com/lepture/authlib/blob/master/authlib/integrations/flask_helpers.py

Reply:
As said in this [issue](https://github.com/lepture/authlib/issues/156), the RFC 6749 specifies the format 'application/x-www-form-urlencoded'